src/array.cr

```diff
@@ -590,6 +590,51 @@ class Array(T)
     elem
   end
 
+  # Deletes all elements that are within the given range,
+  # returning that elements.
+  # Raises `IndexError` if the index is out of range.
+  #
+  # ```
+  # a = ["ant", "bat", "cat", "dog"]
+  # a.delete_at(1..2)  #=> ["bat", "cat"]
+  # a                  #=> ["ant", "dog"]
+  # a.delete_at(99..100)  #=> IndexError
+  # ```
+  def delete_at(range : Range)
```

Comment:
We restrict `[]` etc to `Int`, I feel like we should do the same here, so `Range(Int, Int)`